addons/web_editor/static/src/js/wysiwyg/plugin/keyboard.js

```diff
@@ -1221,46 +1190,125 @@ var KeyboardPlugin = AbstractPlugin.extend({
+        return {
+            beforePoint: beforePoint,
+            beforeText: previousText,
+            afterPoint: afterPoint,
+            afterText: afterText,
+        };
+    },
+    _setComposition: function (text) {
+        var range = this.context.invoke('editor.createRange');
+        var compo = this._getCompositionWord(range);
+        var length = compo.beforeText.length + compo.afterText.length;
+        var point = compo.beforePoint;
 
-            var range = this.context.invoke('editor.setRange', accentPlaceholder, 0);
+        // complete the word
+        if (!compo.afterText.length && !text.indexOf(compo.beforeText)) {
+            text = text.slice(length);
             range.select();
+        } else {
+            if (length === (text.length + 1) && !compo.beforeText.indexOf(text)) {
+                // long press delete
+                text = '';
+            }
 
-            setTimeout(function () {
-                var accentedChar = accentPlaceholder.innerHTML;
-                $(accentPlaceholder).remove();
-                if (parentContenteditable) {
-                    $parent.attr('contenteditable', parentContenteditable);
-                } else {
-                    $parent.removeAttr('contenteditable');
+            // autocorrect the word (change only the updated char, and keep the format for previous chars)
+            var good = point;
+            var remove = false;
+            while (point && point.node && length) {
+                if (dom.isText(point.node)) {
+                    var char = point.node.textContent.slice(point.offset, point.offset + 1);
+                    if (char.length) {
+                        length--;
+                        if (!remove && text[0] === char) {
+                            text = text.slice(1);
+                            good = point;
+                        } else {
+                            point.node.textContent = point.node.textContent.slice(0, point.offset) + point.node.textContent.slice(point.offset + 1);
+                            remove = true;
+                            continue;
+                        }
+                    }
                 }
-                baseRange.select();
-                self.context.invoke('HelperPlugin.insertTextInline', accentedChar);
-            });
-        } else {
-            this.context.invoke('HelperPlugin.insertTextInline', e.key);
+                point = dom.nextPoint(point);
+            }
+            point = dom.nextPoint(good);
+
+            range.sc = range.ec = point.node;
+            range.so = range.eo = point.offset;
+            range.select();
+        }
+
+        if (text.length) {
+            this.context.invoke('HelperPlugin.insertTextInline', text);
+            var range = this.context.invoke('editor.createRange');
+            var textContent = range.sc.textContent;
+            function fuckGoogleKeyboardWhoInsertCharWithoutEvent () {
```

Comment:
fuckGoogleKeyboardWho**m**Insert**s**CharWithoutEvent

Reply:
:+1:

Reply:
fuckGoogleKeyboard**Which**InsertsCharWithoutEvent

Reply:
or fuckGoogleKeyboard**That**InsertsCharWithoutEvent

Reply:
Now, actuallyt, saying it out loud I realize "that" would be better than "which".